src/index.ts

```diff
@@ -8,56 +8,105 @@ import * as formatter from "./helpers/formatter";
+      description:
+        "Search for Mongoose schemas in Javascript files rather than in Typescript files."
     }),
     project: flags.string({
       char: "p",
-      default: "./",
-      description: "path of tsconfig.json or its root folder"
+      description: `Path of tsconfig.json or its root folder. Defaults to "./"`
+    }),
+    imports: flags.string({
+      char: "i",
+      description: "Custom import statements to ensure your function types are all defined",
+      multiple: true
+    }),
+    config: flags.string({
+      char: "c",
+      description:
+        "Path of mtgen.config.json or its root folder. CLI flag options will take precendence over settings in mtgen.config.json"
+    }),
+    module: flags.boolean({
+      description: `generate interfaces in a "declare module 'mongoose'" block`
     })
   };
 
   // path of mongoose models folder
   static args = [
     {
-      name: "root_path",
+      name: "root-path",
       default: "."
     }
   ];
 
+  private getConfig() {
+    const { flags: cliFlags, args } = this.parse(MongooseTsgen);
+
+    type FlagConfig = Omit<typeof cliFlags, "config" | "output" | "project"> & {
+      output: string;
+      project: string;
+    };
+
+    const configFileFlags: Partial<FlagConfig> = paths.getConfigFromFile(cliFlags.config);
+
+    // remove "config" since its only used to grab the config file
+    delete cliFlags.config;
+
+    // we cant set flags as `default` using the official oclif method since the defaults would overwrite flags provided in the config file.
+    // instead, well just set "output" and "project" as default manually if theyre still missing after merge with configFile.
+    configFileFlags.output = configFileFlags.output ?? "./src/interfaces";
+    configFileFlags.project = configFileFlags.project ?? "./";
+
+    return {
+      flags: {
+        ...configFileFlags,
+        ...cliFlags
+      } as FlagConfig,
+      args
+    };
+  }
+
   async run() {
+    const { flags, args } = this.getConfig();
+
     cli.action.start("Generating mongoose typescript definitions");
-    const { flags, args } = this.parse(MongooseTsgen);
 
-    let fullTemplate: string;
+    if (!flags.module) {
+      const pathSegments = flags.output?.split?.("/");
+      // if no output path (used to default to /src/types/mongoose/index.d.ts) or if output path is a folder path,  warn that this library does not add typescript interfaces as declared modules by default (now requires --module flag)
+      if (
+        !pathSegments ||
+        pathSegments[pathSegments.length - 1].match(/[a-zA-Z0-9_-]*\.[A-Za-z]{2,3}/)
+      ) {
+        this.warn(
+          "⚠ Breaking change in update to v5.1.0 - Run `npx mtgen --help` for more details ⚠\n* Generated types and interfaces are now exported from the generated file, rather than augmented on to the mongoose module. Use `--module` for previous behaviour.\n* The default value for the --output flag has changed."
```

Comment:
```suggestion
          "⚠ Breaking change in update to v5.1.0 - Run `npx mtgen --help` for more details ⚠\n* Generated types and interfaces are now exported from the generated file, rather than augmented on to the mongoose module. Use `--augment` for previous behaviour.\n* The default value for the --output flag has changed."
```